materialize-rockset/driver.go

```diff
@@ -42,9 +44,48 @@ func (c *config) Validate() error {
 }
 
 type resource struct {
-	Workspace    string `json:"workspace,omitempty"`
-	Collection   string `json:"collection,omitempty"`
-	MaxBatchSize int    `json:"maxBatchSize,omitempty"`
+	Workspace  string `json:"workspace,omitempty"`
+	Collection string `json:"collection,omitempty"`
+	// Configures the rockset collection to bulk load an initial data set from an S3 bucket, before transitioning to
+	// using the write API for ongoing data.
+	// If a previous version of this materialization wrote files into S3 in order to more quickly backfill historical
+	// data, then this value should contain configuration about the integration.  See: https://go.estuary.dev/rock-bulk
+	// If a bulk loading integration is not being used, then this should be undefined.
+	InitializeFromS3 *cloudStorageIntegration `json:"initializeFromS3,omitempty"`
+	MaxBatchSize     int                      `json:"maxBatchSize,omitempty"`
+}
+
+// Configuration for bulk loading data into the new Rockset collection from a cloud storage bucket.
+type cloudStorageIntegration struct {
+	// The name of the integration in Rockset, which would have been created in the Rockset UI.
+	Integration string `json:"integration"`
+	// The name of the S3 bucket to load data from.
+	Bucket string `json:"bucket"`
+	// The region of the S3 bucket. Optional.
+	Region string `json:"region"`
```

Comment:
I haven't actually verified this, but I think they may try to discover the region automatically, and this is only required if they don't have permission. I works if I don't specify it with a test bucket is in us-east-2. I'm hesitant to require it, since Rockset doesn't and their users may have the expectation that it's optional.